partitions.go

```diff
@@ -206,8 +206,7 @@ func (p *partitions) getAndCacheActivePartition(rootChecker func(StatCommander,
 func getBootEnvActivePartition(env BootEnvReadWriter) (string, error) {
 	bootEnv, err := env.ReadEnv("mender_boot_part")
 	if err != nil {
-		log.Error(err)
```

Comment:
Why remove the logging?

Reply:
I tried to explain this in the commit message:

> Also reorganize the logging and error returning to include the error when returning higher up in the callstack, otherwise the resulting log entry is quite misleading.

The problem was that the error you get further up in the call stack looks completely out-of-context unless paired with this one, so since we now return the full error message, it makes sense to remove this and let the caller print all of it at once.